server/mlabns/tests/test_tool_fetcher.py

```diff
@@ -87,33 +88,51 @@ def initToolIdSiteGroup(self):
         self.createSliverTool(
             tool_id='mock_tool_a', site_id='abc01', country='CountryA',
             status_ipv4=message.STATUS_ONLINE,
-            status_ipv6=message.STATUS_ONLINE)
+            status_ipv6=message.STATUS_ONLINE,
+            fqdn='mock_tool_a.mlab1.abc01.measurement-lab.org')
+        self.createSliverTool(
+            tool_id='mock_tool_a', site_id='abc01', country='CountryA',
+            status_ipv4=message.STATUS_ONLINE,
+            status_ipv6=message.STATUS_ONLINE,
+            fqdn='mock_tool_a.mlab2.abc01.measurement-lab.org')
         self.createSliverTool(
             tool_id='mock_tool_b', site_id='abc01', country='CountryA',
             status_ipv4=message.STATUS_ONLINE,
-            status_ipv6=message.STATUS_ONLINE)
+            status_ipv6=message.STATUS_ONLINE,
+            fqdn='mock_tool_b.mlab1.abc01.measurement-lab.org')
         self.createSliverTool(
             tool_id='mock_tool_c', site_id='abc01', country='CountryA',
             status_ipv4=message.STATUS_ONLINE,
-            status_ipv6=message.STATUS_ONLINE)
+            status_ipv6=message.STATUS_ONLINE,
+            fqdn='mock_tool_b.mlab1.abc01.measurement-lab.org')
         self.createSliverTool(
             tool_id='mock_tool_a', site_id='abc02', country='CountryA',
             status_ipv4=message.STATUS_ONLINE,
-            status_ipv6=message.STATUS_OFFLINE)
+            status_ipv6=message.STATUS_OFFLINE,
+            fqdn='mock_tool_a.mlab1.abc02.measurement-lab.org')
         self.createSliverTool(
             tool_id='mock_tool_b', site_id='abc02', country='CountryA',
             status_ipv4=message.STATUS_OFFLINE,
-            status_ipv6=message.STATUS_OFFLINE)
+            status_ipv6=message.STATUS_OFFLINE,
+            fqdn='mock_tool_b.mlab1.abc02.measurement-lab.org')
         self.createSliverTool(
             tool_id='mock_tool_a', site_id='abc03', country='CountryA',
             status_ipv4=message.STATUS_OFFLINE,
-            status_ipv6=message.STATUS_OFFLINE)
+            status_ipv6=message.STATUS_OFFLINE,
+            fqdn='mock_tool_a.mlab1.abc03.measurement-lab.org')
         self.createSliverTool(
             tool_id='mock_tool_c', site_id='abc03', country='CountryA',
             status_ipv4=message.STATUS_OFFLINE,
-            status_ipv6=message.STATUS_OFFLINE)
+            status_ipv6=message.STATUS_OFFLINE,
+            fqdn='mock_tool_c.mlab1.abc03.measurement-lab.org')
         self.insertCreatedTools()
 
+    def testOnlyReturnMlab1(self):
+        self.initToolIdSiteGroup()
+        tool_properties = tool_fetcher.ToolProperties(tool_id='mock_tool_a')
+        for tool in self.fetcher.fetch(tool_properties):
+          self.assertTrue("mlab2" not in tool.fqdn)
```

Comment:
I'd prefer to assert that it contains the correct fqdn rather than verifying that it doesn't have the wrong ones. If the filter was broken and was rejecting more than it should, this would still pass.